src/main/java/com/eternalcode/core/chat/ChatManager.java

```diff
@@ -0,0 +1,99 @@
+package com.eternalcode.core.chat;
+
+import com.eternalcode.core.configuration.ConfigurationManager;
+import com.eternalcode.core.configuration.MessagesConfiguration;
+import com.eternalcode.core.configuration.PluginConfiguration;
+import com.eternalcode.core.utils.ChatUtils;
+import com.eternalcode.core.utils.DateUtils;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.papermc.paper.event.player.AsyncChatEvent;
+import lombok.Getter;
+import org.bukkit.Bukkit;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import panda.std.Option;
+
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public final class ChatManager {
+
+    @Getter private final Cache<UUID, Long> chatCache;
+    @Getter private boolean                 enabled;
+    private final char[]                    clearMessage;
+    private final ConfigurationManager      configManager;
+
+    public ChatManager(ConfigurationManager manager){
+        this.configManager = manager;
+
+        this.chatCache = CacheBuilder.newBuilder()
+            .expireAfterWrite(configManager.getPluginConfiguration().chatSlowMode * 1000L, TimeUnit.SECONDS)
+            .build();
+        this.enabled = true;
+        this.clearMessage = new char[30000];
+        Arrays.fill(this.clearMessage, ' ');
+    }
+
+    public void clearChat(CommandSender sender){
+        MessagesConfiguration messages = this.configManager.getMessagesConfiguration();
+        Bukkit.getOnlinePlayers().forEach(player -> {
+            player.sendMessage(String.valueOf(clearMessage));
+            Bukkit.broadcast(ChatUtils.component(messages.chatCleared.replace("{NICK}", sender.getName())));
+
+        });
+    }
+
+    public void switchChat(CommandSender sender, boolean statue){
+        MessagesConfiguration messages = this.configManager.getMessagesConfiguration();
+        if (statue){
+            if (this.enabled){
+                sender.sendMessage(ChatUtils.color(messages.chatAlreadyEnabled));
+                return;
+            }
+            this.enabled = true;
+            Bukkit.broadcast(ChatUtils.component(messages.chatEnabled.replace("{NICK}", sender.getName())));
+            return;
+        }
+        if (!this.enabled){
+            sender.sendMessage(ChatUtils.color(messages.chatAlreadyDisabled));
+            return;
+        }
+        this.enabled = false;
+        Bukkit.broadcast(ChatUtils.component(messages.chatDisabled.replace("{NICK}", sender.getName())));
+    }
+
+    public void slowMode(CommandSender sender, String slowMode){
+        MessagesConfiguration messages = this.configManager.getMessagesConfiguration();
+        PluginConfiguration config = this.configManager.getPluginConfiguration();
+        if (!ChatUtils.isNumber(slowMode)){
```

Comment:
ogólnie co do stylu kodu to raczej nie ściskamy go, i dodajemy entry przed if i po.